proof_test.go

```diff
@@ -1088,3 +1088,40 @@ func TestGenerateProofWithOnlyAbsentKeys(t *testing.T) {
 		}
 	}
 }
+
+func TestProofOfPresenceWithEmptyValue(t *testing.T) {
```

Comment:
This test case is pretty simple, and I believe shows a potentially fundamental flaw of the optimization done compared to the spec reference regarding "leaving out" the extension statuses for stems that have both proof of presence and absence.

The test case is simple:
- Add a key with a value.
- Prove that another key that corresponds to that `LeafNode` has value `nil`.

This fails for what looks like a simple to fix bug, but after you do the obvious fix it will be apparent that that will break another test and the fundamental issue will become more apparent. 

I can explain more in Matrix.